WcaOnRails/app/controllers/api/internal/v1/payment_controller.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+class Api::Internal::V1::PaymentController < Api::Internal::V1::ApiController
+  def init
+    competition_id, user_id = params["attendee_id"].split("-")
+
+    amount = params["amount"].to_i
```

Comment:
Passing around `amount`s is tricky because of https://stripe.com/docs/currencies#special-cases that sometimes don't comply with the ISO standards that our RubyMoney gem is trying hard to use.

We have a dedicated `StripeTransaction.amount_to_stripe` and `.amount_to_ruby` helper duo which works as long as we can safely assume that all Ruby amounts come from the RubyMoney gem. Now that they come from the external world, this assumption might break.

Not necessarily requiring a specific change here, just trying to point out to be _very_ careful and explicitly mark amounts as "Ruby amount" or "Stripe amount" at least through a comment.

Reply:
This doesn't necessarily come from the "external world" as it is an internal request, I will make sure to use Ruby Money there, but is there maybe a way we can guarantee that this fulfills some kind of API contract?

Reply:
The only thing close to a contract that we can rely on is ISO 4217. It's the one that defines the currency codes, like `USD` and `EUR`, but actually it also defines sub-units: https://en.wikipedia.org/wiki/ISO_4217#Minor_unit_fractions